mailarchive/mailarchive-subetha/src/java/org/sakaiproject/mailarchive/SakaiMessageHandlerFactory.java

```diff
@@ -172,8 +177,12 @@ public void from(String from) throws RejectException {
             public void recipient(String to) throws RejectException {
                 SplitEmailAddress address = SplitEmailAddress.parse(to);
 
-                if (serverConfigurationService.getServerName().equalsIgnoreCase(address.getDomain())) {
-                    // || serverConfigurationService.getServerNameAliases().contains(address.getDomain())) {
+                if(to.startsWith(serverConfigurationService.getString("msgcntr.messages.header.from.reply"))) {
```

Comment:
Define a constant here and provide spacing in the if.

if (to.startsWith(serverConfigurationService.getString("msgcntr.messages.header.from.reply"))) {

Define a default value in the getString method, something like serverConfigurationService.getString(CONSTANT, StringUtils.EMPTY).

Reply:
Invert the comparison order please.